dipy/viz/ui.py

```diff
@@ -2952,6 +2952,283 @@ def set_img(self, img):
+
+    def _add_to_renderer(self, ren):
+        """ Add all subcomponents or VTK props that compose this UI component.
+
+        Parameters
+        ----------
+        ren : renderer
+        """
+        self.button.add_to_renderer(ren)
+        self.text.add_to_renderer(ren)
+
+    def _get_size(self):
+        width = self.button.size[0] + self.button_label_gap + self.text.size[0]
+        height = max(self.button.size[1], self.text.size[1])
+        return np.array([width, height])
+
+    def _set_position(self, coords):
+        """ Position the lower-left corner of this UI component.
+
+        Parameters
+        ----------
+        coords: (float, float)
+            Absolute pixel coordinates (x, y).
+        """
+        num_newlines = self.label.count('\n')
+        self.button.position = coords + \
+            (0, num_newlines * self.font_size * 0.5)
+        offset = (self.button.size[0] + self.button_label_gap, 0)
+        self.text.position = coords + offset
+
+
+class Checkbox(UI):
+
+    """ A 2D set of :class:'Option' objects.
+    Multiple options can be selected.
+
+    Attributes
+    ----------
+    labels : list(string)
+        List of labels of each option.
+    options : list(Option)
+        List of all the options in the checkbox set.
+    padding : float
+        Distance between two adjacent options
+    """
+
+    def __init__(self, labels, padding=1, font_size=18,
+                 font_family='Arial', position=(0, 0)):
+        """
+        Parameters
+        ----------
+        labels : list(string)
+            List of labels of each option.
+        padding : float
+            The distance between two adjacent options
+        font_size : int
+            Size of the text font.
+        font_family : str
+            Currently only supports Arial.
+        position : (float, float)
+            Absolute coordinates (x, y) of the lower-left corner of
+            the button of the first option.
+        """
+        self.labels = labels
+        self._padding = padding
+        self._font_size = font_size
+        self.font_family = font_family
+        super(Checkbox, self).__init__(position)
+        self.on_change = lambda: None
+        self.checked = []
+
+    def _setup(self):
+        """ Setup this UI component.
+        """
+        self.options = []
+        button_y = self.position[1]
+        for label in self.labels:
+            option = Option(label=label,
+                            font_size=self.font_size,
+                            position=(self.position[0], button_y))
+            line_spacing = option.text.actor.GetTextProperty().GetLineSpacing()
+            button_y = button_y + self.font_size * \
+                (label.count('\n') + 1) * (line_spacing + 0.1) + self.padding
+            option.button.on_left_mouse_button_pressed = self.toggle_check
+            self.options.append(option)
+            option.button.add_callback(option.text.actor,
```

Comment:
I think we can all agree on the way we are handling events is unintuitive (and probably bogus) right? @Garyfallidis @dmreagan @ranveeraggarwal 
In this particular instance, we are adding a callback (which is a method of a **CheckBox** object) to the VTKActor of a **TextBlock2D** object using the add_callback of a **Button2D** object!

I think the proper way would have been something like:
`option.on_change = self._handle_new_selection`.

Reply:
I did this to avoid looping through all the options in the callback. This sends the button of the text clicked as an argument to the `add_callback()` function, so that its icon can be changed.

Reply:
I see. Wouldn't it be better if the `Option` object was handling that for you? To me, whether the `TextBlock2D` is associated with a particular `Button2D` shouldn't influence the behavior of `CheckBox` and `RadioBox` classes.

Reply:
See https://github.com/karandeepSJ/dipy/pull/1 for a way of doing what I described above.